### ISSUE TITLE
Broken Link in development-cycle.md document

### ISSUE THREAD

**Issue by fjbo-net** [cotheca/docs#39]
## Description
When clicking on the `[release cycle document](./releases/release-cycle.md)` link in [development-cycle.md:55](https://github.com/cotheca/docs/blob/2f50c3306743a5416ce7f1b54c5acc41f73a83e8/contributing/development-cycle.md?plain=1#L55C221-L55C274) it causes a not found error. Link address is wrong.

## Expected Behavior
Expected link to redirect the page to the [@cotheca/docs/contributing/release-cycle.md](https://github.com/cotheca/docs/blob/main/contributing/release-cycle.md) file.

### Objectives
Please specify what a valid fix should accomplish (one or more objectives):
- [x] Create a new branch for the issue based on 'main'
- [x] Fix [link](https://github.com/cotheca/docs/blob/2f50c3306743a5416ce7f1b54c5acc41f73a83e8/contributing/development-cycle.md?plain=1#L55C221-L55C274) address to `[release cycle document](./release-cycle.md)`
- [x] Find and update (if any) other links to old sections, like:
  - `./development/**/*.*`
  - `./releases/**/*.*`
  - `./development/releases/**/*.*`
- Submit a Pull Request to merge into 'main'
- [x] After successful PR merge, report new changes to #38, so that branch 'wip-38-SyncSeparationOfConcernsWithMain' can sync new changes from main and propagate changes into branch 'wip-7-SeparationOfConcerns' after merging 'wip-38-SyncSeparationOfConcernsWithMain' into 'wip-7-SeparationOfConcerns' without additional issues or Pull Requests.
  - [x] (1.) Submit a Pull Request for #39
  - [x] (2.) Notify #38 to sync from 'main' after merging PR corresponding to #39

## Actual Behavior
The link takes the user to the wrong destination `@cotheca/docs/contributing/releases/release-cycle.md`

# Reproducible in
GitHub, clicking on the [mentioned link](https://github.com/cotheca/docs/blob/2f50c3306743a5416ce7f1b54c5acc41f73a83e8/contributing/development-cycle.md?plain=1#L55C221-L55C274) in the [development-cycle.md](https://github.com/cotheca/docs/blob/main/contributing/development-cycle.md) file on any UI.

## Steps to Reproduce
Please provide detailed steps for reproducing the issue.

1. **Open** the [development-cycle.md](https://github.com/cotheca/docs/blob/main/contributing/development-cycle.md) document in GitHub
3. **Click** on the "release cycle document" link under the Approach > The Scrum Side > Sprints paragraph

## Screenshots or Log Output
![image](https://github.com/cotheca/docs/assets/28407576/0273800c-c4b8-4931-b44a-2edf72788864)

**Comment by fjbo-net** [comment]
Blocks:
- #38

**Comment by fjbo-net** [comment]
Blocked by:
- #38

**Comment by fjbo-net** [comment]
No longer a blocker for:
- #38
